### PR TITLE
add an option to not exclude pinned pkgs

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by h-vetinari** [comment]
Thanks. I was looking at the code now, but wouldn't have been able to come up with this right away.

File: conda_forge_tick/migrators/migration_yaml.py [hunk 1]
```diff
@@ -517,13 +517,15 @@ def create_rebuild_graph(
...
 ) -> nx.DiGraph:
     total_graph = copy.deepcopy(gx)
     excluded_feedstocks = set() if excluded_feedstocks is None else excluded_feedstocks
-    # Always exclude the packages themselves from the migration
-    for node in package_names:
-        excluded_feedstocks.update(gx.graph["outputs_lut"].get(node, {node}))
+    # Exclude the packages themselves from the migration
+    if exclude_pinned_pkgs:
+        for node in package_names:
+            excluded_feedstocks.update(gx.graph["outputs_lut"].get(node, {node}))
```
Comment (h-vetinari) [inline conda_forge_tick/migrators/migration_yaml.py:528]:
```suggestion
    # Generally, the packages themselves should be excluded from the migration;
    # an example for exceptions are migrations for new python versions or architectures,
    # where e.g. numpy needs to be rebuilt despite being pinned.
    if exclude_pinned_pkgs:
        for node in package_names:
            excluded_feedstocks.update(gx.graph["outputs_lut"].get(node, {node}))
```

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
Merge when you are ready @isuruf!

**Comment by beckermr** [comment]
@isuruf This PR is killing the bot due to a bug here: https://github.com/regro/autotick-bot/runs/4082186511?check_suite_focus=true

Can you fix it?

**Comment by beckermr** [comment]
NVM I think I got it.

**Comment by isuruf** [comment]
Thanks for fixing it

**Comment by h-vetinari** [comment]
Sorry for overlooking the typo as well...

**Pull request merged**